lib/pages/splash_page/splash_page_state_notifier.dart

```diff
@@ -15,35 +21,39 @@ class SplashPageStateNotifier extends BaseStateNotifier<SplashPageState> {
     required AuthenticationService authenticationService,
     required RemoteConfigService remoteConfigService,
     required HabitDailySummaryService habitDailySummaryService,
+    required SharedPreferenceService sharedPreferenceService,
     required DeepLinkService deepLinkService,
     required TadabburService tadabburService,
     required this.navigatorKey,
   })  : _authenticationService = authenticationService,
         _remoteConfigService = remoteConfigService,
         _habitDailySummaryService = habitDailySummaryService,
         _tadabburService = tadabburService,
+        _sharedPreferenceService = sharedPreferenceService,
         _deepLinkService = deepLinkService,
         super(
           SplashPageState(),
         );
 
   final AuthenticationService _authenticationService;
+  final SharedPreferenceService _sharedPreferenceService;
   final RemoteConfigService _remoteConfigService;
   final DeepLinkService _deepLinkService;
   final TadabburService _tadabburService;
   final HabitDailySummaryService _habitDailySummaryService;
   final GlobalKey<NavigatorState> navigatorKey;
   final DbLocal db = DbLocal();
 
-  @override
-  Future<void> initStateNotifier() async {
-    final Connectivity conn = Connectivity();
-    final ConnectivityResult res = await conn.checkConnectivity();
+  late AppUpdateType? appUpdateType;
 
+  @override
+  Future<void> initStateNotifier({
+    ConnectivityResult? conn,
+  }) async {
     await _deepLinkService.init(navigatorKey);
     await _authenticationService.initRepository();
 
-    if (res != ConnectivityResult.none) {
+    if (conn != null && conn != ConnectivityResult.none) {
```

Comment:
baru baca documentasi di code nya, connectivity ini ada comment kayak gini yum 
"Do not use the result of this function to decide whether you can reliably, make a network request. It only gives you the radio status."

pun kalo ini bisa dipake untuk ngecheck network mungkin checkernya bisa diubah jadi conn == ConnectivityResult.wifi || conn == ConnectivityResult.mobile. Soalnya bakal ada case misal dia ga connect internet tapi connect bluetooth nilainya bukan none cmiiw. Mungkin bisa jadi bahan pertimbangan

Reply:
nice catch ndras, ini hrsnya kita perlu refactor bwt yg lain jg nih, tp kita jadiin debt dlu yak biar PR nya gak kegedean. Gue akan bikin extension function ajaa bwt handle ini. Otw

Reply:
ini nanti kita memang perlu update & research lebih jauh sih idealnya kita pake connectivity checker ini gimana. Soalnya gua sempet baca dia sebenernya ada listener yg bisa dipake gitu, cm blm sempet kita implement